### PR TITLE
3 parse mag data

### LINKED ISSUE #3: Parse MAG data
MAG data is stored in `pickle` files and then parsed to a PostgreSQL DB.

### PR DESCRIPTION
**Pull request by kstathou**
Parses data collected from MAG API.

Closes #3 .

### CODE REVIEW

**Comment by kstathou** [comment]
Hi both, I followed Alex's advice and I think that the [instructions](https://github.com/nestauk/fnf/tree/3-parse-mag-data/fnf#how-to-collect-sample-data-from-mag-to-run-parse_magpy) and the [documentation as code](https://github.com/nestauk/fnf/blob/3-parse-mag-data/fnf/sample_mag_data.py) should make this easier to review.

**Comment by mindrones** [comment]
Fantastic thanks! I've created this to keep track: https://github.com/nestauk/imt_playbook/issues/28

File: fnf/README.md [hunk 1]
```diff
@@ -32,3 +32,21 @@ test_postgresdb = postgres+psycopg2://postgres@localhost/postgres
...
 `disinfo`: the project's database.  
 `postgres`: default database that is shipped with PostgreSQL and used here for testing the ORMs.
+
+## How to collect sample data from MAG to run `parse_mag.py` ##
+Instead of getting all the data for a MAG Field of Study by running the [`query_fos_mag.py`](https://github.com/nestauk/fnf/blob/dev/fnf/query_fos_mag.py), it's preferable to get a small subset in order to test that the MAG response is loaded, parsed and stored in `postgres` correctly.
+
+You should do the following:
+
+``` python
+python sample_mag_data.py
```
Comment (bishax) [inline fnf/README.md:42]: :+1: This helps for reviewing

We should probably think about  adding in test modes for cases like this, but we would need to think about the cleanest way to do it.

File: fnf/data/parse_mag_data.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    """Parse paper information from a MAG API response.
+    
+    Args:
+        response (json): Response from MAG API in JSON format. Contains paper information.
+
+    Returns:
+        d (dict): Paper metadata.
+
+    """
+    d = {}
```
Comment (bishax) [inline fnf/data/parse_mag_data.py]: We could reduce repetition here:
```
remappings = {"id": "Id", ... "citations": "CC"}
d = {k: response[v] for k, v in remappings.items()}
```

File: fnf/data/parse_mag_data.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    d["id"] = response["Id"]
+    d["prob"] = response["prob"]
+    d["title"] = response["Ti"]
+    d["publication_type"] = response["Pt"]
+    d["year"] = response["Y"]
+    d["date"] = response["D"]
+    d["citations"] = response["CC"]
+    try:
```
Comment (bishax) [inline fnf/data/parse_mag_data.py:32]: A comment here about why the keys may be missing would help understandability

Comment (bishax) [inline fnf/data/parse_mag_data.py:32]: There is quite a lot of repeated logic here

File: fnf/parse_mag.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import logging
```
Comment (bishax) [inline fnf/parse_mag.py:4]: Add a docstring here which lets us know what this script does

**Review by bishax — COMMENTED** [review]
Looking great!
One docstring needed then feel free to merge.

(Whilst cloning and configuring, a few issues came up which I'm about to open a separate issue for as they relate to previous PR's)

**Comment by kstathou** [comment]
Thanks @bishax, made the changes and merging!

**Pull request merged**